.github/workflows/build-container.yml

```diff
@@ -5,6 +5,12 @@ on:
   workflow_dispatch:
   schedule:
     - cron: '0 0 * * 0'
+  push:
+    paths:
+    - 'Dockerfile'
+  pull_request:
+    paths:
+    - 'Dockerfile'
 
 jobs:
   build-and-publish:
```

Comment:
The changed code looks nice, although the last run showed that we should upgrade the checkout action as v2 seems to be deprecated (we are at v4 now)

https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/

https://github.com/ochosi/pr-review-queue/actions/runs/7311557524

Reply:
Makes sense! I've bumped `actions/checkout` to v4 wherever it's used in this repo.